rdflib/graph.py

```diff
@@ -558,7 +558,10 @@ def __isub__(self, other):
     def __add__(self, other):
         """Set-theoretic union
            BNode IDs are not changed."""
-        retval = Graph()
+        try:
+            retval = self.__class__()
```

Comment:
Oh, I do prefer using `type(self)` over `self.__class__`. I actually changed everything to  `self.__class__` after I saw it being used instead of `type(self)` in `term.py`. I was just trying to keep it consistent with rest of the code. I'll make the change right away. :)